File: packages/typescript/src/territorial/generateTerritorialSummary.ts
import type { Intersection, RiskLevel } from "../types";

const ORDER: RiskLevel[] = ["ALTO", "MEDIO", "BAJO"];

export function generateTerritorialSummary(
  intersections: Intersection[]
): string {
  if (intersections.length === 0) {
    return "El territorio no presenta superposiciones con las capas evaluadas.";
  }

  const present = new Set(intersections.map((item) => item.level));
  const dominant = ORDER.find((level) => present.has(level)) ?? "BAJO";

  const entities = intersections
    .map((item) => item.entity ?? "")
    .join(", ");

  return `El territorio presenta superposicion de nivel ${dominant} con ${intersections.length} entidad(es): ${entities}.`;
}

File: packages/python/peru_spatial_utils/territorial/generate_territorial_summary.py
from typing import List, Mapping, Sequence

_PRIORITY = ["ALTO", "MEDIO", "BAJO"]


def generate_territorial_summary(
    intersections: Sequence[Mapping[str, str]],
) -> str:
    if not intersections:
        return "El territorio no presenta superposiciones con las capas evaluadas."

    present = {item.get("level") for item in intersections}
    dominant = next(level for level in _PRIORITY if level in present)
    entities: List[str] = [item.get("entity", "") for item in intersections]
    count = len(intersections)
    joined = ", ".join(entities)
    return (
        f"El territorio presenta superposicion de nivel {dominant} con "
        f"{count} entidad(es): {joined}."
    )
